### ISSUE TITLE
Why Expose / Offer Choice in Bundler?

### ISSUE THREAD

**Issue by jhoffmcd** [sveltejs/sapper#1178]
I have a question which may also evolve into a proposal. I was wondering why expose the bundler configuration or even offer an option between Rollup and Webpack?

To explain the motivation behind my question, I am coming from using Gatsby.js and Next.js pretty heavily for the last 2 or 3 years. Both of those frameworks obfuscate the bundler/transpiler configuration from the user **until they need to extend it**. They also keep it opinionated by sticking to one bundler (webpack). I think this has maybe in a small way helped the adoption rate of these frameworks by making them more accessible to developers of all skill levels.

While I like the fact that I have a choice to use Rollup or Webpack, why offer both, and why expose the configuration it if the user shouldn't have to (or maybe doesn't want to) mess with the configuration? Maybe keep both options, but normalize the config with a `sapper.config.js` or something along those lines, where you can choose `bundler: "rollup"`?

When I introduce Next.js or Gastby.js to new development teams, they love the fact that it enables them to get to feature development asap. I really see the potential with Sapper to do the same but even more efficiently.

**Comment by Conduitry** [comment]
Sapper originally only supported webpack because Rollup's codesplitting wasn't quite in place yet. Later, once it was, Rollup supported was added, and there was no particularly compelling reason to remove webpack.

In Sapper, the bundler configuration isn't so horrifically complicated that there's much of a compelling reason to hide that either. And there are rather compelling reasons not to hide it. Almost every project of a certain size will require some sort of customization in its bundler configuration. There's then pressure for Sapper/create-react-app/whatever to provide a configuration option and to increase its API surface and increase the amount of things that need to be documented and increase the number of places that things can go wrong - or, if the project decides not to let whatever it is be directly customizable, users are stuck with ejecting their configuration, leaving them with lots of code they don't need and shouldn't have to worry about understanding because it was written to be as general and all-encompassing as possible.

Finally, if directly exposing people to the bundler configuration they're using forces them to be a little more familiar with it, that's a good thing. I have a thing or two to say about webpack's mammoth API - and how this has led to too many people giving up, accepting that bundler configurations are incomprehensible, and finding them so scary as to wish they were hidden away in node_modules with all the other scary things.

In short: The bundler configuration being visible is a feature.
